### PR TITLE
bugfix: add zk serializer.

### PR DESCRIPTION
**Pull request by objcoding**
### Ⅰ. Describe what this PR did
add zk serializer.

### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/seata/seata/issues/2337

### CODE REVIEW

**Comment by a364176773** [comment]
server/src/main/resources/registry.conf need to add?

File: config/seata-config-zk/src/main/java/io/seata/config/zk/ZookeeperConfiguration.java [hunk 1]
```diff
@@ -245,4 +250,23 @@ public void handleDataDeleted(String s) {
...
+    private ZkSerializer getZkSerializer() {
+        ZkSerializer zkSerializer = null;
+        String serializer = FILE_CONFIG.getConfig(FILE_CONFIG_KEY_PREFIX + SERIALIZER_KEY);
+        if (StringUtils.isNotBlank(serializer)) {
+            try {
+                Class<?> aClass = Class.forName(serializer);
+                zkSerializer = (ZkSerializer) aClass.newInstance();
+            } catch (ClassNotFoundException cfe) {
+                LOGGER.error("No zk serializer class found, serializer:{}", serializer, cfe);
+            } catch (Exception e) {
```
Comment (jsbxyyx) [inline config/seata-config-zk/src/main/java/io/seata/config/zk/ZookeeperConfiguration.java:262]: Exception replace to Throwable ? Class.forName can be appears LinkageError

Comment (objcoding) [reply config/seata-config-zk/src/main/java/io/seata/config/zk/ZookeeperConfiguration.java:264]: Done.

**Comment by objcoding** [comment]
> registry.c

Yes, it needs to be added here.

File: server/src/main/resources/registry.conf [hunk 2]
```diff
@@ -54,7 +54,7 @@ registry {
 
 config {
   # file、nacos 、apollo、zk、consul、etcd3
-  type = "file"
+  type = "zk"
```
Comment (l81893521) [inline server/src/main/resources/registry.conf:57]: Default config type is file.

Comment (objcoding) [reply server/src/main/resources/registry.conf]: Done.

File: config/seata-config-zk/src/main/java/io/seata/config/zk/ZookeeperConfiguration.java [hunk 3]
```diff
@@ -247,4 +252,23 @@ public void handleDataDeleted(String s) {
...
+                Class<?> aClass = Class.forName(serializer);
+                zkSerializer = (ZkSerializer) aClass.newInstance();
+            } catch (ClassNotFoundException cfe) {
+                LOGGER.error("No zk serializer class found, serializer:{}", serializer, cfe);
+            } catch (Throwable cause) {
+                LOGGER.error("found zk serializer encountered an unknown exception", cause);
+            }
+        }
+        if (zkSerializer == null) {
+            zkSerializer = new DefaultZkSerializer();
```
Comment (l81893521) [inline config/seata-config-zk/src/main/java/io/seata/config/zk/ZookeeperConfiguration.java:269]: How about use log to tell users, zkSerializer would use DefaultZkSerializer

Comment (objcoding) [reply config/seata-config-zk/src/main/java/io/seata/config/zk/ZookeeperConfiguration.java:269]: Done.

File: config/seata-config-core/src/main/resources/registry.conf [hunk 4]
```diff
@@ -66,6 +66,7 @@ config {
     serverAddr = "127.0.0.1:2181"
     sessionTimeout = 6000
     connectTimeout = 2000
+    serializer = ""
```
Comment (slievrly) [inline config/seata-config-core/src/main/resources/registry.conf:69]: It is not recommended that the serializer be exposed to the user to increase the cost of understanding what code is actually defined for the configuration item.

Comment (objcoding) [reply config/seata-config-core/src/main/resources/registry.conf]: This suggestion is very good.

**Pull request merged**